x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/snapshotlifecycle/action/GetSnapshotLifecycleAction.java

```diff
@@ -0,0 +1,201 @@
+
+    protected GetSnapshotLifecycleAction() {
+        super(NAME);
+    }
+
+    @Override
+    public GetSnapshotLifecycleAction.Response newResponse() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Writeable.Reader<GetSnapshotLifecycleAction.Response> getResponseReader() {
+        return GetSnapshotLifecycleAction.Response::new;
+    }
+
+    public static class Request extends AcknowledgedRequest<GetSnapshotLifecycleAction.Request> {
+
+        private String[] lifecycleIds;
+
+        Request(String... lifecycleIds) {
+            this.lifecycleIds = Objects.requireNonNull(lifecycleIds, "ids may not be null");
+        }
+
+        Request() {
+            this.lifecycleIds = Strings.EMPTY_ARRAY;
+        }
+
+        public String[] getLifecycleIds() {
+            return this.lifecycleIds;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            lifecycleIds = in.readStringArray();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeStringArray(lifecycleIds);
+        }
+
+        @Override
+        public int hashCode() {
+            return Arrays.hashCode(lifecycleIds);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == null) {
+                return false;
+            }
+            if (obj.getClass() != getClass()) {
+                return false;
+            }
+            Request other = (Request) obj;
+            return Arrays.equals(lifecycleIds, other.lifecycleIds);
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private List<SnapshotLifecyclePolicyItem> lifecycles;
+
+        public Response() { }
+
+        public Response(List<SnapshotLifecyclePolicyItem> lifecycles) {
+            this.lifecycles = lifecycles;
+        }
+
+        public Response(StreamInput in) throws IOException {
+            this.lifecycles = in.readList(SnapshotLifecyclePolicyItem::new);
+        }
+
+        @Override
+        public String toString() {
+            return Strings.toString(this);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            for (SnapshotLifecyclePolicyItem item : lifecycles) {
+                builder.startObject(item.getPolicy().getId());
+                builder.field("version", item.getVersion());
+                builder.field("modified_date", item.getModifiedDate());
+                builder.field("policy", item.getPolicy());
+                builder.endObject();
```

Comment:
Why not put this in a `toXContent` method on `SnapshotLifecyclePolicyItem`?

Reply:
Sure I can do that.